src/modules/udw/services/bookmark.service.js

```diff
@@ -0,0 +1,106 @@
+import { showErrorNotification } from '../../common/services/notification.service';
+import {
+    getBasicRequestInit,
+    handleRequestResponse,
+    handleRequestError,
+    handleRequestResponseStatus,
+} from '../../common/helpers/request.helper.js';
+
+const ENDPOINT_BOOKMARK = '/api/ezp/v2/bookmark';
+
+/**
+ * Loads bookmarks
+ *
+ * @function loadBookmarks
+ * @param {Object} restInfo REST config hash containing: token and siteaccess properties
+ * @param {Function} callback
+ */
+export const loadBookmarks = (restInfo, limit, offset, callback) => {
+    const basicRequestInit = getBasicRequestInit(restInfo);
+    const request = new Request(`${ENDPOINT_BOOKMARK}?limit=${limit}&offset=${offset}`, {
+        ...basicRequestInit,
+        method: 'GET',
+        headers: {
+            ...basicRequestInit.headers,
+            Accept: 'application/vnd.ez.api.ContentTypeInfoList+json',
+        },
+    });
+
+    fetch(request)
+        .then(handleRequestResponse)
+        .then(callback)
+        .catch(showErrorNotification);
+};
+
+/**
+ * Adds bookmark
+ *
+ * @function addBookmark
+ * @param {Object} restInfo REST config hash containing: token and siteaccess properties
+ * @param {String} locationId location id
+ * @param {Function} callback
+ */
+export const addBookmark = (restInfo, locationId, callback) => {
+    const basicRequestInit = getBasicRequestInit(restInfo);
+    const request = new Request(`${ENDPOINT_BOOKMARK}/${locationId}`, {
+        ...basicRequestInit,
+        method: 'POST',
+    });
+
+    fetch(request)
+        .then(handleRequestResponseStatus)
+        .then(callback)
+        .catch(showErrorNotification);
+};
+
+/**
+ * Removes bookmark
+ *
+ * @function removeBookmark
+ * @param {Object} restInfo REST config hash containing: token and siteaccess properties
+ * @param {String} locationId location id
+ * @param {Function} callback
+ */
+export const removeBookmark = (restInfo, locationId, callback) => {
+    const basicRequestInit = getBasicRequestInit(restInfo);
+    const request = new Request(`${ENDPOINT_BOOKMARK}/${locationId}`, {
+        ...basicRequestInit,
+        method: 'DELETE',
+    });
+
+    fetch(request)
+        .then(handleRequestResponseStatus)
+        .then(callback)
+        .catch(showErrorNotification);
+};
+
+/**
+ * Checks if given location is bookmarked
+ *
+ * @function checkIsBookmarked
+ * @param {Object} restInfo REST config hash containing: token and siteaccess properties
+ * @param {String} locationId location id
+ * @param {Function} callback
+ */
+export const checkIsBookmarked = (restInfo, locationId, callback) => {
+    const basicRequestInit = getBasicRequestInit(restInfo);
+    const request = new Request(`${ENDPOINT_BOOKMARK}/${locationId}`, {
+        ...basicRequestInit,
+        method: 'HEAD',
+    });
+    const bookmarkedStatusCode = 200;
+    const notBookmarkedStatusCode = 404;
```

Comment:
Can we change this and use always 200 with the response `{ bookmarked: true|false}`?

Reply:
Good idea, but I'll keep it like this, Because we need changes on the backend side as well. This work has not been planned yet.